### ISSUE TITLE
C++: constexpr and / or static trailing return type syntax + void

### ISSUE THREAD

**Issue by bat2grb** [doxygen/doxygen#10186]
Hi,

I'm using doxygen version 1.9.1 on Ubuntu 22.04. For 

```cpp
/// The class ...
class FooTrailing {
public:
   /// The function ...
   auto Bar() -> void {}

   /// The inline function ...
   inline auto InlineBar() -> void {}

   /// The constexpr function ...
   constexpr auto ConstexprBar() -> void {}

   /// The static function ...
   static auto StaticBar() -> void {}

   /// The static inline function ...
   static inline auto StaticInlineBar() -> void {}

   /// The static constexpr function ...
   static constexpr auto StaticConstexprBar() -> void {}
};
```

still 

```
example/FooTrailing.h:11: warning: return type of member FooTrailing::ConstexprBar is not documented
example/FooTrailing.h:14: warning: return type of member FooTrailing::StaticBar is not documented
example/FooTrailing.h:17: warning: return type of member FooTrailing::StaticInlineBar is not documented
example/FooTrailing.h:20: warning: return type of member FooTrailing::StaticConstexprBar is not documented
```

is reported. Where as for

```cpp
/// The class ...
class Foo {
public:
   /// The function ...
   void Bar() {}

   /// The inline function ...
   inline void InlineBar() {}

   /// The constexpr function ...
   constexpr void ConstexprBar() {}

   /// The static function ...
   static void StaticBar() {}

   /// The static inline function ...
   static inline void StaticInlineBar() {}

   /// The static constexpr function ...
   static constexpr void StaticConstexprBar() {}
};
```

no warnings at all.

[example.zip](https://github.com/doxygen/doxygen/files/12092497/example.zip)

Partly related to [#6442](https://github.com/doxygen/doxygen/issues/6442)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10187

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
